hknweb/tutoring/forms.py

```diff
@@ -1,83 +1,94 @@
+                label=str(course.course),
+                widget=forms.RadioSelect(choices=COURSE_PREFERENCE_CHOICES),
+            )
             self.fields[field_name].initial = pref.preference
+
     def save_course_preference_data(self):
         for name, value in self.cleaned_data.items():
-            id = int(name.replace('course_preference_', ''))
+            id = int(name.replace("course_preference_", ""))
             course = TutorCourse.objects.get(id=id)
             pref = CoursePreference.objects.get(course=course, tutor=self.tutor)
             pref.preference = value
             pref.save()
 
-SLOT_PREFERENCE_CHOICES= [
-        (0, 'Unavailable'),
-        (1, 'Available'),
-        (2, 'Preferred'),
-    ]
-ADJACENT_PREFERENCE_CHOIES = [
-    (-1, 'No'),
-    (0, "Don't care"),
-    (1, 'Yes')
+
+SLOT_PREFERENCE_CHOICES = [
+    (0, "Unavailable"),
+    (1, "Available"),
+    (2, "Preferred"),
+]
+ADJACENT_PREFERENCE_CHOIES = [(-1, "No"), (0, "Don't care"), (1, "Yes")]
+
+ROOM_PREFERENCE_CHOICES = [
+    (n, "%s %s" % (building, room_num))
+    for n, building, room_num in Room.DEFAULT_ROOM_CHOICES
 ]
+
+
 class TimeSlotPreferenceForm(forms.Form):
-    adjacent_pref = forms.IntegerField(widget=forms.RadioSelect(choices=ADJACENT_PREFERENCE_CHOIES))
+    adjacent_pref = forms.IntegerField(
+        widget=forms.RadioSelect(choices=ADJACENT_PREFERENCE_CHOIES)
+    )
     num_assignments = forms.IntegerField()
     tutor_name = forms.CharField()
+
     def __init__(self, *args, **kwargs):
-        self.tutor = kwargs.pop('tutor')
+        self.tutor = kwargs.pop("tutor")
         super().__init__(*args, **kwargs)
-        self.fields['tutor_name'].initial = self.tutor.name
-        self.fields['adjacent_pref'].initial = self.tutor.adjacent_pref
-        self.fields['num_assignments'].initial = self.tutor.num_assignments
-        timeslots = TimeSlot.objects.all().order_by('timeslot_id')
+        self.fields["tutor_name"].initial = self.tutor.name
+        self.fields["adjacent_pref"].initial = self.tutor.adjacent_pref
+        self.fields["num_assignments"].initial = self.tutor.num_assignments
+        timeslots = TimeSlot.objects.all().order_by("timeslot_id")
         for timeslot in timeslots:
             pref = TimeSlotPreference.objects.get(tutor=self.tutor, timeslot=timeslot)
-            field_name = 'timeslot_time_preference_%s' % (timeslot.timeslot_id,)
-            self.fields[field_name] = forms.IntegerField(widget=forms.RadioSelect(choices=SLOT_PREFERENCE_CHOICES))
+            field_name = "timeslot_time_preference_%s" % (timeslot.timeslot_id,)
+            self.fields[field_name] = forms.IntegerField(
+                widget=forms.RadioSelect(choices=SLOT_PREFERENCE_CHOICES)
+            )
             self.fields[field_name].initial = pref.time_preference
 
-            number_of_tutor_rooms = Room.objects.all().count()
+            field_name = "timeslot_office_preference_%s" % (timeslot.timeslot_id,)
```

Comment:
There is a concern that the algorithm supports two rooms, and not three
We could probably do the Online 9-10 one by hand somehow

Reply:
Followup in https://github.com/compserv/hknweb/pull/324#issuecomment-912884700 since this issue doesn't just pertain to this set of code